src/ostree/ot-builtin-size-summary.c

```diff
@@ -0,0 +1,138 @@
+  gs_unref_variant GVariant *index = NULL;
+  gs_free gchar *refspec = NULL;
+  gs_free gchar *revision = NULL;
+  gint64 archived = 0;
+  gint64 unpacked = 0;
+  gsize  fetch_needed = 0;
+  gint64 new_archived = 0;
+  gint64 new_unpacked = 0;
+
+  context = g_option_context_new("[REMOTE] BRANCH - Display the summary for branch");
+  g_option_context_add_main_entries (context, options, NULL);
+
+  if (!ostree_option_context_parse (context, options, &argc, &argv, OSTREE_BUILTIN_FLAG_NONE, &repo, cancellable, error))
+    goto out;
+
+  if (!ostree_repo_open (repo, cancellable, error))
+    goto out;
+
+  switch (argc)
+    {
+    case 3:
+      remote = argv[1];
+      ref = argv[2];
+      refspec = g_strdup_printf ("%s/%s", remote, ref);
+      break;
+
+    case 2:
+      ref = argv[1];
+      refspec = g_strdup (ref);
+      break;
+
+    case 1:
+      ot_util_usage_error (context, "BRANCH must be specified", error);
+      goto out;
+
+    default:
+      ot_util_usage_error (context, "only one branch may be summarised", error);
+      goto out;
+    }
+
+  // if there's a remote, we might not have pulled the metadata yet:
+  if (!ostree_repo_resolve_rev (repo, refspec, remote != NULL, &revision, error))
+    goto out;
+
+  // nothing in the cache, but try and fetch it if it's a remote refspec:
+  if (!revision && remote)
+    {
+      OstreeRepoPullFlags flags = OSTREE_REPO_PULL_FLAGS_METADATA;
+      gchar *pullrefs[] = { (gchar *) ref, NULL };
+
+      if (!ostree_repo_pull (repo, remote, pullrefs, flags, NULL, cancellable, error))
+        goto out;
+
+      if (!ostree_repo_resolve_rev (repo, refspec, FALSE, &revision, error))
+        goto out;
+    }
+
+  if (!revision)
+    {
+      if (!*error)
+        g_set_error (error, G_IO_ERROR, G_IO_ERROR_NOT_FOUND,
+                     "Refspec '%s' not found", refspec);
+      goto out;
+    }
+
+  if (ostree_repo_get_commit_sizes (repo, revision,
+                                          &new_archived,
+                                          &new_unpacked,
+                                          &fetch_needed,
+                                          &archived, &unpacked,
+                                          &entries,
+                                          cancellable, error))
+    {
+      g_print ("Summary for refspec %s:\n"
+               "  files: %" G_GSIZE_FORMAT "/%" G_GSIZE_FORMAT " entries\n"
+               "  archived: %" G_GINT64_FORMAT "/%" G_GINT64_FORMAT "\n"
+               "  unpacked: %" G_GINT64_FORMAT "/%" G_GINT64_FORMAT "\n",
+               refspec,
+               entries - fetch_needed, entries,
+               archived - new_archived, archived,
+               unpacked - new_unpacked, unpacked);
+    }
+  else if (!*error) // No error found, 0 entries => no index available
+    {
+      g_set_error (error, G_IO_ERROR, G_IO_ERROR_NOT_FOUND,
+                   "Refspec '%s' has no index, cannot summarise", refspec);
+      goto out;
```

Comment:
Whoops, that was left over from when `ostree_repo_get_commit_sizes` returned something different.